spec/unit/event-timeline-set.spec.ts

```diff
@@ -206,85 +206,97 @@ describe("EventTimelineSet", () => {
             expect(liveTimeline.getEvents().length).toStrictEqual(0);
         });
 
-        it("should allow edits to be added to thread timeline", async () => {
-            jest.spyOn(client, "supportsThreads").mockReturnValue(true);
-            jest.spyOn(client, "getEventMapper").mockReturnValue(eventMapperFor(client, {}));
-            Thread.hasServerSideSupport = FeatureSupport.Stable;
+        describe("When the server supports threads", () => {
+            let previousThreadHasServerSideSupport: FeatureSupport;
 
-            const sender = "@alice:matrix.org";
+            beforeAll(() => {
+                previousThreadHasServerSideSupport = Thread.hasServerSideSupport;
+                Thread.hasServerSideSupport = FeatureSupport.Stable;
```

Comment:
an alternative to stashing the previous value might be to do some `jest.spyOn`, which will naturally be reset after the tests?

Reply:
I tried to do this with `jest.spyOn` and couldn't get it to work, so merging as-is for now.